File: Portfolio/Portfolio.Blazor/Data/Entities/ResourceLink.cs
using System.ComponentModel.DataAnnotations;

namespace Portfolio.Blazor.Data.Entities;

public sealed class ResourceLink
{
    public int Id { get; set; }
    public int ResourceId { get; set; }

    [Required(ErrorMessage = "URL is required.")]
    public string Url { get; set; } = string.Empty;
    public string? Label { get; set; }
    public string? Platform { get; set; }
    public string? Purpose { get; set; }
    public bool IsPrimary { get; set; }
    public bool IsFree { get; set; }
    public int? LanguageId { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Resource? Resource { get; set; }
    public Language? Language { get; set; }
}

File: src/laravel/app/Models/ResourceLink.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ResourceLink extends Model
{
    use HasFactory;

    protected $fillable = ['resource_id', 'url', 'label', 'platform', 'purpose', 'is_primary', 'is_free', 'language_id'];

    protected $casts = ['is_primary' => 'boolean', 'is_free' => 'boolean'];

    /**
     * @return BelongsTo<resource, $this>
     */
    public function resource(): BelongsTo
    {
        return $this->belongsTo(Resource::class);
    }

    /**
     * @return BelongsTo<Language, $this>
     */
    public function language(): BelongsTo
    {
        return $this->belongsTo(Language::class);
    }
}
